packages/wonder-blocks-color/util/utils.test.js

```diff
@@ -0,0 +1,190 @@
+// @flow
+import {fade, mix} from "./utils.js";
+
+const INVALID_COLORS = ["#NOTVALID", "#12345", "#12", "#ABCDEFG", ""];
+
+describe("mix", () => {
+    test("color is not a string, throws", () => {
+        // Arrange
+        const color = ((null: any): string);
+
+        // Act
+        const underTest = () => mix(color, "#FFF");
+
+        // Assert
+        expect(underTest).toThrowErrorMatchingSnapshot();
+    });
+
+    test("background is not a string, throws", () => {
+        // Arrange
+        const background = ((null: any): string);
+
+        // Act
+        const underTest = () => mix("#FFF", background);
+
+        // Assert
+        expect(underTest).toThrowErrorMatchingSnapshot();
+    });
+
+    test("color is not a valid string format, throws", () => {
+        // Arrange
+        const testpoints = INVALID_COLORS;
+
+        // Act
+        for (const testpoint of testpoints) {
+            const underTest = () => mix(testpoint, "#FFF");
+            expect(underTest).toThrowErrorMatchingSnapshot();
+        }
+    });
+
+    test("background is not a valid string format, throws", () => {
+        // Arrange
+        const testpoints = INVALID_COLORS;
+
+        // Act
+        for (const testpoint of testpoints) {
+            const underTest = () => mix("#FFF", testpoint);
+            expect(underTest).toThrowErrorMatchingSnapshot();
+        }
+    });
+
+    test("color has no alpha, returns color with background alpha", () => {
+        // Arrange
+        const testpoints = [
+            ["#000", "#abc", "#000000"],
+            ["#FFFFFF", "rgba(10,20,30,0.5)", "rgba(255,255,255,0.50)"],
+            ["rgb(200,200,190)", "rgba(0,0,0,0.23)", "rgba(200,200,190,0.23)"],
+            ["rgba(0,0,0,1)", "#1D3baa", "#000000"],
+            ["#ABC", "rgba(0,0,0,1)", "#aabbcc"],
+        ];
+
+        for (const [color, background, expectation] of testpoints) {
+            // Act
+            const result = mix(color, background);
+
+            // Assert
+            expect(result).toBe(expectation);
+        }
+    });
+
+    test("color has alpha, returns mix with background alpha", () => {
+        // Arrange
+        const testpoints = [
+            ["rgba(100,100,100,0.2)", "#FFF", "#e0e0e0"],
+            ["rgba(100,100,100,0.2)", "rgba(0,0,0,0.5)", "rgba(20,20,20,0.50)"],
+            ["rgba(100,100,100,0.2)", "rgba(0,0,0,0.5)", "rgba(20,20,20,0.50)"],
```

Comment:
I would expect the alpha to be higher, but since we don't use this anywhere I okay to leave this as a `TODO`.

Reply:
This was the existing behavior. I didn't change that as `mix` seemed to be more explicit about how it handled alpha, whereas fade implied it did something that it didn't, which I then fixed.

Reply:
That said, I notice that I duplicated a test point here. Should probably fix that.